python/sparkdl/graph/input.py

```diff
@@ -0,0 +1,232 @@
+                              feed_names=None,
+                              fetch_names=None):
+        """
+        Construct a TFInputGraphBuilder from a model checkpoint
+        """
+        assert (feed_names is None) == (fetch_names is None), \
+            'feed_names and fetch_names, if provided must appear together'
+        assert (feed_names is None) != (signature_def_key is None), \
+            'must either provide feed_names or singnature_def_key'
+
+        def import_graph_fn(sess):
+            # Load checkpoint and import the graph
+            with sess.as_default():
+                ckpt_path = tf.train.latest_checkpoint(checkpoint_dir)
+
+                # NOTE(phi-dbq): we must manually load meta_graph_def to get the signature_def
+                #                the current `import_graph_def` function seems to ignore
+                #                any signature_def fields in a checkpoint's meta_graph_def.
+                meta_graph_def = meta_graph_pb2.MetaGraphDef()
+                with open("{}.meta".format(ckpt_path), 'rb') as fin:
+                    meta_graph_def.ParseFromString(fin.read())
+
+                saver = tf.train.import_meta_graph(meta_graph_def, clear_devices=True)
+                saver.restore(sess, ckpt_path)
+
+                sig_def = None
+                if signature_def_key is not None:
+                    sig_def = meta_graph_def.signature_def[signature_def_key]
+                    assert sig_def, 'singnature_def_key {} provided, '.format(signature_def_key) + \
+                        'but failed to find it from the meta_graph_def ' + \
+                        'from checkpoint {}'.format(checkpoint_dir)
+
+            return _GinBuilderInfo(sig_def=sig_def)
+
+        return _GinBuilder(import_graph_fn).build(feed_names, fetch_names)
+
+    @classmethod
+    def _from_saved_model_impl(cls, saved_model_dir, tag_set,
+                               signature_def_key=None,
+                               feed_names=None,
+                               fetch_names=None):
+        """
+        Construct a TFInputGraphBuilder from a SavedModel
+        """
+        assert (feed_names is None) == (fetch_names is None), \
+            'feed_names and fetch_names, if provided must appear together'
+        assert (feed_names is None) != (signature_def_key is None), \
+            'must either provide feed_names or singnature_def_key'
+
+        def import_graph_fn(sess):
+            tag_sets = tag_set.split(',')
+            meta_graph_def = tf.saved_model.loader.load(sess, tag_sets, saved_model_dir)
+
+            sig_def = None
+            if signature_def_key is not None:
+                sig_def = tf.contrib.saved_model.get_signature_def_by_key(
+                    meta_graph_def, signature_def_key)
+
+            return _GinBuilderInfo(sig_def=sig_def)
+
+        return _GinBuilder(import_graph_fn).build(feed_names, fetch_names)
+
+
+class _GinBuilderInfo(object):
```

Comment:
this seems to be really more like _SigDefInfo - let's name it more clearly.

Reply:
Removed all classes and builder objects.